tests/integration-tests/conftest.py

```diff
@@ -691,3 +693,24 @@ def architecture(request, instance, region):
 def key_name(request):
     """Return the EC2 key pair name to be used."""
     return request.config.getoption("key_name")
+
+
+@pytest.fixture()
+def pcluster_ami_without_standard_naming(region, os, architecture, request):
+    """
+    Define a fixture to manage the creation and deletion of AMI without standard naming.
+
+    This AMI is used to test the validation of pcluster version in Cookbook
+    """
+    ami_id = None
+
+    def _pcluster_ami_without_standard_naming(version):
+        nonlocal ami_id
+        ami_id = retrieve_pcluster_ami_without_standard_naming(region, os, version, architecture)
+        return ami_id
+
+    yield _pcluster_ami_without_standard_naming
+
+    if ami_id and not request.config.getoption("no_delete"):
```

Comment:
I don't think we want to apply the no_delete here since there is not a good way to reuse this resource in a next round of tests.

Reply:
With the AMI created, the test code can reuse the AMI in the next round of test and save 4 minutes of copying the AMI

Reply:
that makes sense. still leaving undeleted resources that are defined outside of a CloudFormation stack, when specifying the --no-delete flag, makes it harder to perform a clean-up afterwards. For now I'd be in favor of only using the no-delete for vpc and cluster stacks and keep its behavior simple enough, with the downside of waiting 4 extra minutes when debugging this test. What do you think?